crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -303,14 +307,33 @@ impl<'db> CallBindingError<'db> {
     ) {
         match self {
             Self::InvalidArgumentType {
+                callable_ty,
                 parameter,
                 argument_index,
                 expected_ty,
                 provided_ty,
             } => {
+                let mut messages = vec![];
+                if let Some(func_lit) = callable_ty.into_function_literal() {
+                    let func_lit_scope = func_lit.body_scope(context.db());
+                    let mut span = Span::from(func_lit_scope.file(context.db()));
+                    let node = func_lit_scope.node(context.db());
+                    if let Some(func_def) = node.as_function() {
+                        let range = argument_index
+                            .and_then(|i| func_def.parameters.iter().nth(i))
+                            .map(|param| param.range())
+                            .unwrap_or(func_def.parameters.range);
+                        span = span.with_range(range);
+                        messages.push(SecondaryDiagnosticMessage::new(
+                            span,
+                            "function defined here",
```

Comment:
I think something like this might be a better message? Since the range highlights the parameter declaration rather than the function definition

```suggestion
                            "parameter declared here",
```

or:

```suggestion
                            "parameter declared in function definition here",
```